apps/api/src/queries/buildViewModelForResource/viewModels/vocabulary-list.view-model.ts

```diff
@@ -49,8 +57,68 @@ export class VocabularyListEntryViewModel extends BaseDomainModel {
 
         this.variableValues = { ...variableValues };
     }
+
+    public canUserWithGroups(userWithGroups: CoscradUserWithGroups) {
+        return (
+            this.term.isPublished || this.term.accessControlList.canUserWithGroups(userWithGroups)
+        );
+    }
 }
 
+/**
+ * TODO If we like this approach, let's move code from here into a test util.
+ * Also note that this has implications for our approach to generating standard
+ * `OpenApi` schemas, including samples.
+ */
+const CanonicalResourceView = (
```

Comment:
I like this. The only question is if we can find a way to incrementally "strangle out" the old pattern.

Reply:
Note that this has now been cleaned up, tested, and moved to a proper utils directory.